plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/ServerSettingsWizardPageModel.java

```diff
@@ -472,7 +472,7 @@ public void setRoute(IRoute newRoute) {
 				this.deployProject, this.projects, 
 				this.sourcePath, this.podPath, this.useInferredPodPath,
 				getService(), getServiceItems(),
-				route, this.selectDefaultRoute, this.routesByProject,
+				newRoute, this.selectDefaultRoute, this.routesByProject,
```

Comment:
I think it's very strange that getRoutes() returns all routes, but it is just a model, and so it's not worth changing. I'd suggesting adding a new proper api method with javadoc that is getRoutesForService()  with no argument, and possible a second with an IService attribute, which will return only the routes relevant to the selected (or passed in) service.